components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/api/CibaAuthServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.ciba.api;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth.ciba.common.AuthReqStatus;
+import org.wso2.carbon.identity.oauth.ciba.common.CibaConstants;
+import org.wso2.carbon.identity.oauth.ciba.dao.CibaDAOFactory;
+import org.wso2.carbon.identity.oauth.ciba.exceptions.CibaCoreException;
+import org.wso2.carbon.identity.oauth.ciba.model.CibaAuthCodeDO;
+import org.wso2.carbon.identity.oauth.ciba.model.CibaAuthCodeRequest;
+import org.wso2.carbon.identity.oauth.ciba.model.CibaAuthCodeResponse;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.TimeZone;
+import java.util.UUID;
+
+/**
+ * Provides authentication services.
+ */
+public class CibaAuthServiceImpl implements CibaAuthService {
+
+    private static Log log = LogFactory.getLog(CibaAuthServiceImpl.class);
+
+    @Override
+    public CibaAuthCodeResponse generateAuthCodeResponse(CibaAuthCodeRequest cibaAuthCodeRequest)
+            throws CibaCoreException {
+
+        CibaAuthCodeDO cibaAuthCodeDO = generateCibaAuthCodeDO(cibaAuthCodeRequest);
+        persistCibaAuthCode(cibaAuthCodeDO);
+        return buildAuthCodeResponse(cibaAuthCodeRequest, cibaAuthCodeDO);
+    }
+
+    /**
+     * Returns a unique AuthCodeKey.
+     *
+     * @return String Returns random uuid.
+     */
+    private static String generateAuthCodeKey() {
+
+        return UUID.randomUUID().toString();
+    }
+
+    /**
+     * Returns a unique auth_req_id.
+     *
+     * @return String Returns random uuid.
+     */
+    private static String generateAuthRequestId() {
+
+        return UUID.randomUUID().toString();
+    }
+
+    /**
+     * Process and return the expires_in for auth_req_id.
+     *
+     * @param cibaAuthCodeRequest Accumulating validated parameters from CibaAuthenticationRequest.
+     * @return long Returns expiry_time of the auth_req_id.
+     */
+    private static long getExpiresIn(CibaAuthCodeRequest cibaAuthCodeRequest) {
+
+        long requestedExpiry = cibaAuthCodeRequest.getRequestedExpiry();
+        if (requestedExpiry == 0) {
+            return CibaConstants.EXPIRES_IN_DEFAULT_VALUE_IN_SEC;
+        } else if (requestedExpiry < CibaConstants.MAXIMUM_REQUESTED_EXPIRY_IN_SEC) {
+            return requestedExpiry;
+        } else {
+            log.warn("(requested_expiry) exceeds default maximum value for the CIBA authentication request made by: " +
```

Comment:
Let's add the requested expiry value here.
So we can have something along the lines of,

"Request Expiry: 2000 seconds exceeds the maximum allowed value: 1000 seconds. Setting the request expirty to maxium allowed value: 1000 seconds."